.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -26,6 +26,4 @@ SV_REPO=
 SV_BRANCH=
 SU_REPO=
 SU_BRANCH=
-LIBHTP_REPO=
```

Comment:
This is my big question.

What do we want to do for hosting libhtp-rs ? Separate repo or inside suricata ?
And if in a private repo, do we want to keep the logic for suricata PRs ?

Reply:
I don't think there is any reason for a private repo, but there are a few options to consider:

1. Separate repo, released as a crate. Typical of many Rust libraries. But this might not be ideal given the tight coupling of Suricata and libhtp, especially around security releases as it works best with independent release cycles. Suricata consuming an unreleased crate would not be straightforward. So think about how this would impact the release process.

2. Separate crate, but inside the `rust/` directory. There are many single repos that contain multiple crates, especially when coupling is required. @libhtp@ could be versioned with Suricata, or still contain its own version cycle, but it lives right inside Suricata, simplifying the release process.  Slightly harder to release to crates.io as it has to deal with some autojunk junk like our current Suricata crates do. If we were to break out certain protocols parsers as their own crates, this is how I would see it happening, rather than moving to their own repos.
    - Questions with this method:
      - Could libhtp still have its own CVEs and security advisories given it lives within the Suricata repo?

Then probably a variety of other build time bundling like we do now and on the fly patching of Cargo.toml.

Reply:
Another option is to do what we do now, git clone it as an extern step then in Cargo.toml:
```
htp = { path = "../libhtp-rs" }
```

Reply:
I agree with you Jason, but I do not know which is better between 1 and 2...

Reply:
1. Is good if we are OK with separate release cycles. For example, would we be OK with releasing libhtp-rs and publishing to crates.io before releasing in Suricata. The best example is to treat it like any other Rust dependency we have, and break the tight coupling, and use Cargo and crates.io as designed.
2. or 3. are better if we need to maintain that coupling.

Reply:
Probably a discussion @victorjulien and @inashivb should be involved in as well with respect to release process and release tooling.

Reply:
What is option 3 ?

(1 does not look so good to me this way)

Reply:
> What is option 3 ?
> 
> (1 does not look so good to me this way)

What we do now, then in `Cargo.toml` use something like:
```
htp = { path = "../libhtp-rs" }
```

Reply:
1. From the release perspective, we **already** need libhtp to be released in order to bundle it in Suricata. The tool then pulls the libhtp it is told to package by the tag and bundles the respective tagged libhtp in Suricata tarball. Ofc if this is security release, we do it right before Suricata release so there is very small window b/w having libhtp and Suricata released.
Q1: Does publishing to crates.io require extra time in any respect to become visible?

2. From CVE perspective, it would indeed not be as easy to segregate libhtp and Suricata CVEs. 
Q2: Is there any extra value in having separate libhtp CVEs? I thought we just did it because it was a separate project in itself on Github historically..

Reply:
> 1. Q1: Does publishing to crates.io require extra time in any respect to become visible?

Only a few minutes.

But something else to consider by using the crate, the version gets pinned in Cargo.lock, so its harder to achieve always building in CI against its current main branch, that is where our bundling comes in.  But then I have to ask, if we're so tightly coupled that way, maybe it should just be in the Suricata repo.